src/sentry/api/serializers/models/app_platform_event.py

```diff
@@ -1,11 +1,62 @@
 from __future__ import absolute_import
 
+import six
+from time import time
+from uuid import uuid4
+from sentry.utils import json
 
-def app_platform_event(action, install, data, actor=None):
-    return {
-        'action': action,
-        'installation': {
-            'uuid': install.uuid,
-        },
-        'data': data,
-    }
+
+class AppPlatformEvent(object):
+    def __init__(self, resource, action, install, data, actor=None):
+        self.resource = resource
+        self.action = action
+        self.install = install
+        self.data = data
+        self.actor = actor
+
+    def get_actor(self):
+        # when sentry auto assigns, auto resolves, etc.
+        # or when an alert rule is triggered
+        if not self.actor:
+            return {
+                'type': 'application',
+                'id': 'sentry',
+                'name': 'Sentry',
+            }
+
+        if self.actor.is_sentry_app:
+            return {
+                'type': 'application',
+                'id': self.install.sentry_app.uuid,
+                'name': self.install.sentry_app.name,
+            }
+
+        return {
+            'type': 'user',
+            'id': self.actor.id,
+            'name': self.actor.name,
+        }
+
+    @property
+    def body(self):
+        return {
+            'action': self.action,
+            'installation': {
+                'uuid': self.install.uuid,
+            },
+            'data': self.data,
+            'actor': self.get_actor(),
+        }
+
+    @property
+    def headers(self):
+        body = json.dumps(self.body)
+        request_uuid = uuid4().hex
+
+        return {
+            'Content-Type': 'application/json',
+            'Request-ID': request_uuid,
```

Comment:
If this is always a pseudo-random value what the use case for consumers? Right now it wouldn't let them de-dupe re-deliveries or retries.

Reply:
Right now, nothing. Eventually this will be passed through the app from endpoints and/or recorded for debugging/other purposes. We just wanted to preemptively add the attribute now.